pkg/steps/artifacts_test.go

```diff
@@ -447,3 +454,70 @@ func TestTestCaseNotifier_SubTests(t *testing.T) {
 		})
 	}
 }
+
+type testPodClient struct {
+	coreclientset.PodsGetter
+	namespace, name string
+}
+
+func (c testPodClient) Exec(namespace, name string, opts *coreapi.PodExecOptions) (remotecommand.Executor, error) {
+	if namespace != c.namespace {
+		return nil, fmt.Errorf("unexpected namespace: %q", namespace)
+	}
+	if name != c.name {
+		return nil, fmt.Errorf("unexpected name: %q", name)
+	}
+	return &testExecutor{command: opts.Command}, nil
+}
+
+type testExecutor struct {
+	command []string
+}
+
+func (e testExecutor) Stream(opts remotecommand.StreamOptions) error {
+	if reflect.DeepEqual(e.command, []string{"tar", "czf", "-", "-C", "/tmp/artifacts", "."}) {
+		var tar []byte
+		tar, err := base64.StdEncoding.DecodeString(`
+H4sIAMq1b10AA+3RPQrDMAyGYc09hU8QrCpOzuOAKR2y2Ar0+HX/tnboEErhfRbxoW8QyEvzwS8uO4r
+dNI63qXOK96yP/JRELZnNdpySSlTrBQlxz6Netua5hiDLctrOa665tA+9Ut9v/pr3/x9+fQQAAAAAAA
+AAAAAAAAAA4GtXigWTnQAoAAA=`)
+		if err != nil {
+			return err
+		}
+		_, err = opts.Stdout.Write(tar)
+		return err
+	} else if reflect.DeepEqual(e.command, []string{"rm", "-f", "/tmp/done"}) {
+		return nil
+	}
+	return fmt.Errorf("unexpected command: %v", e.command)
+}
+
+func TestArtifactWorker(t *testing.T) {
+	tmp, err := ioutil.TempDir("", "")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tmp)
```

Comment:
handle the error:

```go
defer func() {
    if err := os.RemoveAll(tmp); err != nil {
        t.Errorf("couldn't clean up tmpdir: %v", err)
    }
}()
```

Reply:
Copy-pasted, `grep` says we have a bunch of other tests to fix =)